lte/gateway/python/magma/pipelined/app/classifier.py

```diff
@@ -273,104 +341,139 @@ def delete_tunnel_flows(self, i_teid:int, ue_ip_adr:IPAddress,
-
-            flows.delete_flow(self._datapath, self.tbl_num, match)
+        # Delete flow for mtr port
+        self._delete_downlink_tunnel_flows(ue_ip_adr, self.config.mtr_port)
 
+        if ue_ip_adr.version == IPAddress.IPV4:
             # Delete ARP flow for LOCAL port
-            if ue_ip_adr.version == IPAddress.IPV4:
-                match = MagmaMatch(eth_type=ether_types.ETH_TYPE_ARP,
-                                   in_port=self._uplink_port,
-                                   arp_tpa=ipaddress.IPv4Address(ue_ip_adr.address.decode('utf-8')))
-
-            flows.delete_flow(self._datapath, self.tbl_num, match)
+            self._delete_downlink_arp_flows(ue_ip_adr, self._uplink_port)
 
             # Delete ARP flow for mtr port
-            if ue_ip_adr.version == IPAddress.IPV4:
-                match = MagmaMatch(eth_type=ether_types.ETH_TYPE_ARP,
-                                   in_port=self.config.mtr_port,
-                                   arp_tpa=ipaddress.IPv4Address(ue_ip_adr.address.decode('utf-8')))
-
-            flows.delete_flow(self._datapath, self.tbl_num, match)
+            self._delete_downlink_arp_flows(ue_ip_adr, self.config.mtr_port)
 
         return True    
 
-
     def _resume_tunnel_flows(self, precedence:int, i_teid:int,
-                              ue_ip_adr:IPAddress=None):
+                             ue_ip_adr:IPAddress):
         priority = Utils.get_of_priority(precedence)
-        # discard uplink Tunnel
+        # Forward flow for gtp port
         match = MagmaMatch(tunnel_id=i_teid, in_port=self.config.gtp_port)
 
         flows.add_resubmit_next_service_flow(self._datapath, self.tbl_num, match,
                                              priority=priority + 1,
                                              reset_default_register=False,
                                              resubmit_table=self.next_table)
 
-        # discard downlink Tunnel for LOCAL port
-        if not ue_ip_adr:
-            self.logger.error("ue_ip_address is None")
-            return
-        else:
-            ip_match_out = get_ue_ip_match_args(ue_ip_adr, Direction.IN)
-            match = MagmaMatch(eth_type=get_eth_type(ue_ip_adr),
+        # Forward flow for LOCAL port
+        ip_match_out = get_ue_ip_match_args(ue_ip_adr, Direction.IN)
+        match = MagmaMatch(eth_type=get_eth_type(ue_ip_adr),
                            in_port=self._uplink_port, **ip_match_out)
 
-            flows.add_resubmit_next_service_flow(self._datapath, self.tbl_num, match,
-                                                 priority=priority + 1,
-                                                 reset_default_register=False,
-                                                 resubmit_table=self.next_table)
+        flows.add_resubmit_next_service_flow(self._datapath, self.tbl_num, match,
+                                             priority=priority + 1,
```

Comment:
can you remind me why the +1 here again?

Reply:
priority+1 changes done in flowdl PR (#5685).